plugins/nanospell/plugin.js

```diff
@@ -590,24 +594,37 @@
 			 for a given range, get the unique words in it that we don't have a spellcheck status for
 			 */
 			function scanWordsInRange(range) {
-				var block;
-				var iterator = range.createIterator();
+				var combinedText = '',
+					block,
+					blockList = [],
+					iterator = range.createIterator();
 				while (( block = iterator.getNextParagraph() )) {
 					block.setCustomData('spellCheckInProgress', true);
-					var unknownWords = getUnknownWords(block.getText());
-					startCheckOrMarkWords(unknownWords, block);
+					combinedText += block.getText() + ' ';
```

Comment:
cool, in this case, tis is sufficient for now.  I had some other ideas for structure (like an object where the keys are the block ids) but I don't think there is any current benefit.